cardinal/ecs/world.go

```diff
@@ -222,11 +222,8 @@ func (w *World) ListQueries() []IQuery {
 	return w.registeredQueries
 }
 
-func (w *World) ListTransactions() ([]transaction.ITransaction, error) {
-	if !w.isTransactionsRegistered {
-		return nil, errors.New("cannot list transactions until transaction registration occurs")
-	}
-	return w.registeredTransactions, nil
+func (w *World) ListTransactions() []transaction.ITransaction {
```

Comment:
An empty list is indicative of the same thing. 

Instead of `if err != nil` we can do `if len(txs) == 0` no information is lost here. 

I remove the error because I encountered tests where no transactions are registered. This was causing failed tests due to the bug fix. 

So I just made a design decision that games without transactions can exist (test games for example) so I don't need to throw a explicit error. Like I said in the description I don't care about the directions we go here. 

Either that failed test is wrong or a game with no transactions is wrong. I'm ok with either interpretation.

Reply:
lmk which you prefer I can go that route.

Reply:
The `w.implWorld.LoadGameState()` marks the transactions as "loaded" (even if there are 0 of them). That call happens in StartGame just after your new code. If you move LoadGameState up your tests should pass without modification.

I'd prefer to leave the error in.